yt_dlp/extractor/youtube.py

```diff
@@ -4267,13 +4289,18 @@ def _extract_from_playlist(self, item_id, url, data, playlist, ytcfg):
         playlist_url = urljoin(url, try_get(
             playlist, lambda x: x['endpoint']['commandMetadata']['webCommandMetadata']['url'],
             compat_str))
-        if playlist_url and playlist_url != url:
+
+        # Some playlists are unviewable but YouTube still provides a link to the (broken) playlist page [1]
+        # [1] MLCT, RLTDwFCb4jeqaKWnciAYM-ZVHg
+        is_known_unviewable = re.fullmatch(r'MLCT|RLTD[\w-]{22}', playlist_id)
```

Comment:
doesn't this have to be expanded to cover all the variants (like we discussed on discord for `playnext`)?

Reply:
currently the check above this covers those - youtube doesn't provide a url to the playlist page for them. though we are trusting youtube here lol.

this is just for the case where youtube (incorrectly) does provide a /playlist url.

Reply:
we could use the same check for playnext. Better than nothing even if the match is not comprehensive